### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by germangelv**
## Summary

- Configured authentication functionality
    - src/Startup.cs
    - src/Repositories/Database/UnitOfWork.cs
    - src/Repositories/ApplicationUserRepository.cs
    - src/Repositories/Interfaces/IUnitOfWork.cs
    - src/Repositories/Interfaces/IApplicationUserRepository.cs 
- Added new Controllers, Models and Views for authentication functionality
    - src/Controllers/AccountController.cs
    - src/Controllers/UserManagementController.cs
    - src/Models/ApplicationUser.cs
    - src/Models/Views/LoginViewModel.cs
    - src/Models/Views/MovieViewModel.cs
    - src/Models/Views/RegisterViewModel.cs
    - src/Models/Views/UserViewModel.cs
    - src/Models/Views/UserManagementViewModel.cs
    - src/Views/Account/AccessDenied.cshtml
    - src/Views/Account/Login.cshtml
    - src/Views/Account/Register.cshtml
    - src/Views/Shared/_Layout.cshtml
    - src/Views/Shared/_LoginPartial.cshtml
    - src/Views/UserManagement/Users.cshtml
- Modified other details
    - src/Views/Movie/Edit.cshtml
    - src/Views/Movie/Index.cshtml
- Added new globalizing elements
    - src/Resources/en-US.json
    - src/Resources/es-ES.json
    - src/Resources/fr-FR.json

## Trello Card

https://trello.com/c/IKqAkWN5
	
## Screenshot

- Home View
image
- Register View
image
- Login View
image
- Started session
image
- Access to Movies
image
- Access to registered users
image

### CODE REVIEW

File: src/Controllers/UserManagementController.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        
+        public UserManagementController(IUnitOfWork unitOfWork)
+        {
+            this._unitOfWork = unitOfWork;
+        }
+
+        private IUnitOfWork UnitOfWork
+        {
+            get { return this._unitOfWork; }
+        }
```
[the remark this answers is not in the document]
Comment (germangelv) [reply src/Controllers/UserManagementController.cs]: ok

File: src/Controllers/AccountController.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Authorization;
+using TrainingNet.Models;
+using TrainingNet.Models.Views;
+
```
Comment (truccomarcos) [inline src/Controllers/AccountController.cs]: Remove this empty line

File: src/Controllers/AccountController.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+namespace TrainingNet.Controllers
+{
+  [Route("[controller]"), Authorize]
+    public class AccountController : Controller
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
```
Comment (truccomarcos) [inline src/Controllers/AccountController.cs]: Add an empty line between attributes an constructor

File: src/Controllers/AccountController.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        public SignInManager<ApplicationUser> SignInManager
+        {
+            get { return _signInManager; }
+        }
```
Comment (truccomarcos) [inline src/Controllers/AccountController.cs]: Add an empty line to separate the constructors

Comment (germangelv) [reply src/Controllers/AccountController.cs]: ok

File: src/Controllers/AccountController.cs (hunk 3 above)
Comment (germangelv) [reply src/Controllers/AccountController.cs]: ok

File: src/Controllers/AccountController.cs (hunk 2 above)
Comment (germangelv) [reply src/Controllers/AccountController.cs]: ok

**Pull request merged**